release/cut-official-release.sh

```diff
@@ -0,0 +1,338 @@
+    cat > "${INSTRUCTIONS}" <<- EOM
+Success!  You must now do the following:
+
+EOM
+  fi
+
+  local -r github="https://github.com/kubernetes/kubernetes.git"
+  declare -r DIR="/tmp/kubernetes-${release_type}-release-${new_version}-$(date +%s)"
+  echo "Cloning from '${github}'..."
+  git clone "${github}" "${DIR}"
+
+  # !!! REMINDER !!!
+  #
+  # Past this point, you are dealing with a different clone of the repo at some
+  # version. Don't assume you're executing code from the same repo as this script
+  # is running in. This is a version agnostic process.
+  pushd "${DIR}"
+
+  if [[ "${release_type}" == 'alpha' ]]; then
+    local -r ancestor="v${version_major}.${version_minor}.0-alpha.$((${version_alpha_rev}-1))"
+
+    git checkout "${git_commit}"
+    verify-at-git-commit "${git_commit}"
+    verify-ancestor "${ancestor}"
+
+    alpha-release "${new_version}"
+  elif [[ "${release_type}" == 'beta' ]]; then
+    local -r release_branch="release-${version_major}.${version_minor}"
+    local -r ancestor="v${version_major}.${version_minor}.${version_patch}-beta.$((${version_beta_rev}-1))"
+
+    git checkout "${release_branch}"
+    verify-at-git-commit "${git_commit}"
+    verify-ancestor "${ancestor}"
+
+    beta-release "${new_version}"
+  elif [[ "${release_type}" == 'official' ]]; then
+    local -r release_branch="release-${version_major}.${version_minor}"
+    local -r beta_version="v${version_major}.${version_minor}.$((${version_patch}+1))-beta"
+    local -r ancestor="${new_version}-beta"
+
+    git checkout "${release_branch}"
+    verify-at-git-commit "${git_commit}"
+    verify-ancestor "${ancestor}"
+
+    official-release "${new_version}"
+    beta-release "${beta_version}"
+  else # [[ "${release_type}" == 'series' ]]
+    local -r release_branch="release-${version_major}.${version_minor}"
+    local -r alpha_version="v${version_major}.$((${version_minor}+1)).0-alpha.0"
+    local -r beta_version="v${version_major}.${version_minor}.0-beta"
+    # NOTE: We check the second alpha version, ...-alpha.1, because ...-alpha.0
+    # is the branch point for the previous release cycle, so could provide a
+    # false positive if we accidentally try to release off of the old release
+    # branch.
+    local -r ancestor="v${version_major}.${version_minor}.0-alpha.1"
+
+    git checkout "${git_commit}"
+    verify-at-git-commit "${git_commit}"
+    verify-ancestor "${ancestor}"
+
+    alpha-release "${alpha_version}"
+
+    echo "Branching ${release_branch}."
+    git checkout -b "${release_branch}"
+    versionize-docs-and-commit "${release_branch}"
+
+    beta-release "${beta_version}"
+    git-push ${release_branch}
+  fi
+
+  echo
+  cat "${INSTRUCTIONS}"
+}
+
+function usage() {
+  echo "Usage: ${0} <version> <githash>"
+  echo
+  echo "See docs/devel/releasing.md for more info."
+}
+
+function check-prereqs() {
+  SED=sed
+  if which gsed &>/dev/null; then
+    SED=gsed
+  fi
+  if ! ($SED --version 2>&1 | grep -q GNU); then
```

Comment:
This is copy-pasted from the old script.  Looking at it, I think this is the only way to take into account the possibility that GNU sed is masquerading as `sed`, (and not installed as gsed,) and the possibility that both `sed` and `gsed` are installed as themselves.